### DISCUSSION TITLE
event.context and firebase

### DISCUSSION THREAD

**Discussion by howbizarre** [nuxt/nuxt#17468] [Questions] [answered]
> Develop on Windows with VSCode
> Deploy to Firebase (v 9.7.0)
> Nuxt 3.0.0-rc.3

I'm trying to made a simple API route guard based on cookie.
I have a server middleware witch add a cookie to the context.
```
// server/middleware/userCheck.ts
export default defineEventHandler((event) => {
    event.context.uc = useCookies(event);
});
```

And in a server api check the context
```
// server/api/draw.ts
export default defineEventHandler((event) => {
    return event.context?.uc?.userCookie ? drawAPI : { "Error": "Api require authentication?!" };
});
```
In client:
```
const userCookie = useCookie("userCookie");
onAuthStateChanged(auth, (user) => {
    userCookie.value = user;
});
``` 

I've used Nitro preset to build project.
```
npx cross-env NITRO_PRESET=firebase npm run build
```
And everything works fine on localhost, but on Firebase hosting the middleware stopped processing cookie.
Do you have any idea why everything is OK locally but on Firebase - doesn't?

**Comment by Diizzayy** [comment, marked as answer]
@howbizarre  I suspect that the issue you're facing is due to Firebase Hosting security measures. When paired with Cloud Run or Cloud Functions (nuxt 3 serverless function), Firebase Hosting only permits the `__session` cookie (hence any other cookie is stripped from incoming requests). You can read more about this behavior here - [Firebase Hosting Cookie Behavior](https://firebase.google.com/docs/hosting/manage-cache#using_cookies)

If this is in fact the issue you're having, simply renaming your cookie from `userCookie` to `__session` should suffice

**Reply by howbizarre** [reply]
@Diizzayy Everything working fine now on the Firebase. Thank you.
